### PR TITLE
Feat: Support Linux

### PR DESCRIPTION
**Pull request by ueman**
## :scroll: Description
This adds support for Linux (tested on Ubuntu) and the corresponding example.
This branch currently targets https://github.com/getsentry/sentry-dart/pull/389 because it is a follow up.

## :green_heart: How did you test it?
The current tests make already sure it works. Linux has no native integration yet, so there are no native integrations to test.

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps
- On Linux we don't have any kind of OS and device information

### CODE REVIEW

File: flutter/example/linux/flutter/CMakeLists.txt
Comment (ueman) [inline flutter/example/linux/flutter/CMakeLists.txt:9]: This was autogenerated with this todo

File: flutter/example/linux/flutter/generated_plugin_registrant.cc
Comment (ueman) [inline flutter/example/linux/flutter/generated_plugin_registrant.cc]: There are a couple generated files. They aren't gitignored according to the template. I'm not sure yet wether we can ignore them.

File: flutter/linux/sentry_flutter_plugin.cc
Comment (ueman) [inline flutter/linux/sentry_flutter_plugin.cc:25]: I think we can add support for https://github.com/getsentry/sentry-native here.

File: flutter/example/linux/flutter/generated_plugin_registrant.cc
Comment (ueman) [reply flutter/example/linux/flutter/generated_plugin_registrant.cc]: They can't be ignored

File: .github/workflows/flutter.yml
Comment (marandaneto) [inline .github/workflows/flutter.yml:102]: the other builds have a `--release`, do we need this here?

File: flutter/linux/sentry_flutter_plugin.cc
Comment (marandaneto) [reply flutter/linux/sentry_flutter_plugin.cc:25]: yep, let's not do it, for now, let's just be sure that this plugin does nothing instead of throwing unless it never gets called.

File: flutter/pubspec.yaml
Comment (marandaneto) [inline flutter/pubspec.yaml:45]: should we maybe name all the plugins a bit more specific to the platform? like SentryFlutterLinux or, to rename the others to be always generic? eg SentryFlutterPlugin

File: flutter/linux/sentry_flutter_plugin.cc
Comment (ueman) [reply flutter/linux/sentry_flutter_plugin.cc:25]: it never get's called at all.

Comment (bruno-garcia) [reply flutter/linux/sentry_flutter_plugin.cc:25]: It's great this uses CMake so it should be easy to add sentry-native later.

File: flutter/pubspec.yaml
Comment (bruno-garcia) [reply flutter/pubspec.yaml:45]: If these names are just for internal use (I believe that's the case) it would be beneficial to align on the side of appending the platform so make it easier to navigate through the project

**Review by bruno-garcia — COMMENTED** [review]
Exciting stuff!

File: flutter/pubspec.yaml
Comment (ueman) [reply flutter/pubspec.yaml:45]: Yep, these are internal names. They however don't necessarily map as is to a class or file name. On linux there is for example a struct named after this, but the file is named `sentry_flutter_plugin.cc` and `sentry_flutter_plugin.h`.
Also MacOS and iOS use the same file.

We can align these name but I don't really think it matters that much, because the files are also in their corresponding platform folders. Your call :D

File: .github/workflows/flutter.yml
Comment (ueman) [reply .github/workflows/flutter.yml:102]: Yep good catch

File: flutter/pubspec.yaml
Comment (marandaneto) [reply flutter/pubspec.yaml:45]: its just a matter of consistency, not really "necessary", I agree with @bruno-garcia that it helps to navigate thru the project, even more, when you take care of more repos, which does the same thing, with similar names

File: .github/workflows/flutter.yml
Comment (ueman) [inline .github/workflows/flutter.yml:102]:
```suggestion
            flutter build linux --release
```

**Review by marandaneto — APPROVED** [review]
nice, thanks for doing this, LGTM

**Comment by marandaneto** [comment]
@ueman see https://github.com/getsentry/sentry-dart/pull/407#issuecomment-815997737

File: .github/workflows/flutter.yml
Comment (marandaneto) [inline .github/workflows/flutter.yml:57]: is this really necessary? I don't see as a requirement here https://flutter.dev/docs/get-started/install/linux#install-flutter-manually

Comment (ueman) [reply .github/workflows/flutter.yml:57]: The first few builds failed because at least ninja was not available on the ubuntu image. I then looked up what the Flutter Plus Plugin maintainers did. I haven't yet tried to reduce the number of installed packages.

I think we can drop at least `sudo apt install -y network-manager upower` but I have yet to try it.

Comment (marandaneto) [reply .github/workflows/flutter.yml:57]: it was just curiosity tbh, if the flutter plus plugin does the same, let's keep like this :)
I've not installed any of those on my ubuntu machine and it worked out ootb, but I might have them installed already

Comment (ueman) [reply .github/workflows/flutter.yml:57]: I think there's also a difference between Ubuntu Desktop and Ubuntu Server

**Review by bruno-garcia — APPROVED** [review]
Great stuff. I'll continue the Windows PR based on this

**Comment by ueman** [comment]
I'll leave this open until https://github.com/getsentry/sentry-dart/pull/389 is approved and merged because this depends on the other branch.

**Review by marandaneto — APPROVED** [review]
looks realllyy good, thanks.
LGTM

**Comment by marandaneto** [comment]
> I'll leave this open until #389 is approved and merged because this depends on the other branch.

we can also merge everything to feature/support-desktop as the branch's name is called: support desktop

**Pull request merged**
